src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeEventDispatcher.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Diagnostics.Tracing
+{
+#if FEATURE_PERFTRACING
+    internal sealed class EventPipeEventDispatcher
+    {
+        internal static EventPipeEventDispatcher Instance = new EventPipeEventDispatcher();
+
+        private IntPtr m_RuntimeProviderID;
+
+        private bool m_stopDispatchTask;
+        private Task m_dispatchTask = null;
+        private object m_dispatchTaskLock = new object();
+
+        // For now, only one EventListener at a time gets to control what events are enabled.
+        private EventListener m_controllingEventListener = null;
+
+        private EventPipeEventDispatcher()
+        {
+            // Get the ID of the runtime provider so that it can be used as a filter when processing events.
+            m_RuntimeProviderID = EventPipeInternal.GetProvider(RuntimeEventSource.EventSourceName);
+        }
+
+        internal void SendCommand(EventListener eventListener, EventCommand command, bool enable, EventLevel level, EventKeywords matchAnyKeywords)
```

Comment:
This is because there are two ways to disable:

1. Call ```EventListener.DisableEvents```.
2. Dispose of the ```EventListener```.

I have only tried case 2 so far.  That said, I went ahead and added the code for case 1.